sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,351 @@
+        ((h/4 - lB/2)*omega(t) + (h/4 + lB/2)*omega(t))*A.x
+        >>> A.ang_vel_in(N)
+        omega(t)*N.y
+        >>> B.ang_vel_in(N)
+        omega(t)*N.y + alpha(t)*A.z
+        >>> A.dcm(N)
+        Matrix([
+        [cos(theta(t)), 0, -sin(theta(t))],
+        [            0, 1,              0],
+        [sin(theta(t)), 0,  cos(theta(t))]])
+        >>> B.dcm(N)
+        Matrix([
+        [ cos(phi(t))*cos(theta(t)), sin(phi(t)), -sin(theta(t))*cos(phi(t))],
+        [-sin(phi(t))*cos(theta(t)), cos(phi(t)),  sin(phi(t))*sin(theta(t))],
+        [             sin(theta(t)),           0,              cos(theta(t))]])
+
+    This example can also be done in another way, i.e, without explicitly defining constants,
+    dyamicsymbols, frames etc.
+
+        >>> rod = Body('rod')
+        >>> plate = Body('plate')
+        >>> C = Body('C') #Ceiling
+        >>> lA = (lB - h / 2) / 2 * rod.frame.z
+        >>> lC = (lB/2 + h/4) * rod.frame.z
+        >>> J1 = PinJoint('J1', C, rod, child_joint_pos=lA, parent_axis=C.frame.y, child_axis=rod.frame.y)
+        >>> J2 = PinJoint('J2', rod, plate, parent_joint_pos=lC, parent_axis=rod.frame.z, child_axis=plate.frame.z)
+        >>> J1.kdes()
+        [J1_omega(t) - Derivative(J1_theta(t), t)]
+        >>> J2.kdes()
+        [J2_omega(t) - Derivative(J2_theta(t), t)]
+        >>> rod.masscenter.vel(C.frame)
+        (h/4 - lB/2)*J1_omega(t)*rod_frame.x
+        >>> plate.masscenter.vel(C.frame)
+        ((h/4 - lB/2)*J1_omega(t) + (h/4 + lB/2)*J1_omega(t))*rod_frame.x
+        >>> rod.frame.ang_vel_in(C.frame)
+        J1_omega(t)*C_frame.y
+        >>> plate.frame.ang_vel_in(C.frame)
+        J2_omega(t)*rod_frame.z + J1_omega(t)*C_frame.y
+        >>> rod.frame.dcm(C.frame)
+        Matrix([
+        [cos(J1_theta(t)), 0, -sin(J1_theta(t))],
+        [               0, 1,                 0],
+        [sin(J1_theta(t)), 0,  cos(J1_theta(t))]])
+        >>> plate.frame.dcm(C.frame)
+        Matrix([
+        [ cos(J1_theta(t))*cos(J2_theta(t)), sin(J2_theta(t)), -sin(J1_theta(t))*cos(J2_theta(t))],
+        [-sin(J2_theta(t))*cos(J1_theta(t)), cos(J2_theta(t)),  sin(J1_theta(t))*sin(J2_theta(t))],
+        [                  sin(J1_theta(t)),                0,                   cos(J1_theta(t))]])
+
+    Notes
+    ======
+
+    All symbols and dynamicsymbols used with `Joints` should be used with correct
+    assumption, i.e. positive=True else it may give incorrect results in some cases.
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None, parent_joint_pos=None,
+        child_joint_pos=None, parent_axis=None, child_axis=None):
+
+        super().__init__(name, parent, child, coordinates, speeds, parent_joint_pos, child_joint_pos,
+            parent_axis, child_axis)
+
+    def _generate_coordinates(self, coordinate):
+        coordinates = []
+        if coordinate is None:
+            theta = dynamicsymbols(self._name + '_theta', positive=True)
```

Comment:
No assumptions should be set here, especially not positive. The coordinate can be any real valued number, positive or negative!

Reply:
So shall I set it to `real=True`?